liberapay/i18n/base.py

```diff
@@ -272,6 +272,10 @@ def format_timedelta(self, o, **kw):
     def parse_money_amount(self, string, currency, maximum=D_MAX):
         group_symbol = self.number_symbols['group']
         decimal_symbol = self.number_symbols['decimal']
+        # If first character is relavent currency symbol, remove it and pass only numeric string
+        # InvalidNumber will catch error if using the wrong currency symbol
+        if string[0] == get_currency_symbol(currency):
```

Comment:
This is incorrect and inefficient in several ways:

1. `string[0] == ...` raises an `IndexError` if the string is empty, and creates a new string otherwise.
2. `get_currency_symbol(currency)` boils down to `Locale.parse(LC_NUMERIC).currency_symbols.get(currency, currency)`. The big problem with that is it uses the server's system locale, not the website user's locale. The minor problem is that it unnecessarily creates a transient `Locale` object. This small inefficiency is why Liberapay doesn't use those Babel functions anymore.

The proper and more efficient way do this is:

```suggestion
        if string.startswith(self.currency_symbols.get(currency, currency)):
```

Reply:
ah, good point about the empty string.

And good to know about the Babel functions.